quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java

```diff
@@ -106,56 +179,87 @@ public static ColumnType text() {
-				(statement, position, value) -> statement.setDouble(position, ((Double) value)));
-	}
-	
 	public static ColumnType bigint() {
 		return new ColumnType(ColumnType.Type.BIGINT, false, "bigint", () -> 0L,
 				(statement, position, value) -> statement.setLong(position, ((Number) value).longValue()));
 	}
 
+	public static ColumnType doubles() {
+		return new ColumnType(ColumnType.Type.DOUBLE, false, "double precision", () -> 0.00d,
+				(statement, position, value) -> statement.setDouble(position, (Double) value));
+	}
+
+	public static ColumnType floats() {
+		return new ColumnType(ColumnType.Type.FLOAT, false, "real", () -> 0.00f,
+				(statement, position, value) -> statement.setFloat(position, (Float) value));
+	}
+
+	public static ColumnType numeric(Integer precision, Integer scale) {
+		return new ColumnType(ColumnType.Type.NUMERIC, false, "numeric(" + precision + "," + scale + ")", () -> BigDecimal.valueOf(0.00d),
+				(statement, position, value) -> statement.setBigDecimal(position, new BigDecimal(value.toString())));
+	}
+
+	public static ColumnType numeric(Integer precision) {
+		return new ColumnType(ColumnType.Type.NUMERIC, false, "numeric(" + precision + ")", () -> BigDecimal.valueOf(0.00d),
+				(statement, position, value) -> statement.setBigDecimal(position, new BigDecimal(value.toString())));
+	}
+
+	public static ColumnType numeric() {
+		return new ColumnType(ColumnType.Type.NUMERIC, false, "numeric", () -> BigDecimal.valueOf(0.00d),
+				(statement, position, value) -> statement.setBigDecimal(position, new BigDecimal(value.toString())));
+	}
+
 	public static ColumnType timestamp(boolean withTimezone) {
 		String typeNotation = "timestamp" + (withTimezone ? " with time zone" : "");
 		return new ColumnType(ColumnType.Type.TIMESTAMP, true, typeNotation, () -> new Timestamp(new Date().getTime()),
 				(statement, position, value) -> statement.setTimestamp(position, (Timestamp) value));
 	}
 
+	public static ColumnType timestamp(boolean withTimezone, Integer length) {
+		String typeNotation = "timestamp (" + length + ")" + (withTimezone ? " with time zone" : "");
+		return new ColumnType(ColumnType.Type.TIMESTAMP, true, typeNotation, () -> new Timestamp(new Date().getTime()),
+				(statement, position, value) -> statement.setTimestamp(position, (Timestamp) value));
+	}
+
 	public static ColumnType date() {
 		return new ColumnType(ColumnType.Type.DATE, true, "date", () -> new java.sql.Date(new Date().getTime()),
 				(statement, position, value) -> statement.setDate(position, (java.sql.Date) value));
 	}
 
-	public static ColumnType doubles() {
-		return new ColumnType(ColumnType.Type.DOUBLE, false, "real", () -> 0.00d,
-				(statement, position, value) -> statement.setDouble(position, (Double) value));
-	}
-
-	public static ColumnType floats() {
-		return new ColumnType(ColumnType.Type.FLOAT, false, "double precision", () -> 0.00f,
```

Comment:
These were switched around, "real" is a float and "double precision" is a double